flaxandteal/coral-arches
# Plugin permissions

### How to test

Note that this has a custom `arches` branch - `feature/plugin-permissions`

Make sure you have the Group and Arches Plugin models loaded from this branch. Run `apply_sets`. Create a user group called "All Users" and add the arches plugins for `init-workflow`  to it. Add "All Users" as a member of the "Global Group".

1. Create a Person
2. Add the Person to All Users
3. Open the Person report (_not_ Edit) and, under User Account, click to copy the link that is created (this will create a `registration_links` table entry)
4. Signup with that link a separate browser (e.g. private browsing)
5. Run `apply_sets` again (in reality this is a scheduled delay, so would like to trigger a job here perhaps in subsequent PR)
6. Note that you have Resource Viewer/Editor/Exporter groups and can create resources without them being provisional
7. You should also see the Worklfows on the left-hand side

To ensure other workflows are viewable. Having done the above,

1. Add a workflow to the All Users group under Arches Plugins
2. Run `apply_sets`
3. Log in / refresh the new user and they should be able to access the added Workflow from the Workflows page
